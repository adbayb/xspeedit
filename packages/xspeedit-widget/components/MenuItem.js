import React from "react";
import cn from "classnames";
import PropTypes from "prop-types";

import Icon from "./Icon";

const MenuItem = ({ iconName, url, isActive, onClick }) => {
	const image = <Icon name={iconName} />;

	return (
		<a
			className={cn({ active: isActive })}
			href={url}
			target={url && "_blank"}
			role={url ? undefined : "button"}
			onClick={url ? undefined : onClick}
		>
			{image}
			<style jsx>{`
				a {
					display: flex;
					flex: 1;
					cursor: pointer;
					transition: background-color 500ms;
					justify-content: center;
					align-items: center;
				}
				.active {
					background-color: #34b7b6;
				}
			`}</style>
		</a>
	);
};

MenuItem.propTypes = {
	iconName: PropTypes.string.isRequired,
	isActive: PropTypes.bool,
	url: PropTypes.string,
	onClick: PropTypes.func
};

export default MenuItem;
