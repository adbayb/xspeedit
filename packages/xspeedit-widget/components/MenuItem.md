**Preview:**

```js
<MenuItem
	iconName="github"
	onClick={() => {
		console.log("Menu item clicked");
	}}
/>;
```

**Example:**

```jsx static
import React from "react";

const Example = () => {
	const handleClick = (e) => {
		console.log("Menu item clicked");
	};

	return (
		<MenuItem
			iconName="github"
			onClick={handleClick}
		/>
	);
};
```
