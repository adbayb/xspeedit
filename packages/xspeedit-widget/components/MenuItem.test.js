import React from "react";
import MenuItem from "./MenuItem.js";

describe("MenuItem", () => {
	it("should render correctly when it's not active", () => {
		// GIVEN: notre composant:
		const tree = shallow(<MenuItem iconName="github" onClick={() => {}} />);
		// THEN: il doit s'afficher correctement en matchant son snapshot:
		expect(tree).toMatchSnapshot();
	});

	it("should render correctly when it's active", () => {
		// GIVEN: notre composant:
		const tree = shallow(<MenuItem iconName="github" onClick={() => {}} isActive />);
		// THEN: il doit s'afficher correctement en matchant son snapshot:
		expect(tree).toMatchSnapshot();
	});

	it("should render correctly when it's a link", () => {
		// GIVEN: notre composant:
		const tree = shallow(
			<MenuItem
				iconName="github"
				url="https://github.com/adbayb/xspeedit"
				onClick={() => {}}
				isActive
			/>
		);
		// THEN: il doit s'afficher correctement en matchant son snapshot:
		expect(tree).toMatchSnapshot();
	});
});
