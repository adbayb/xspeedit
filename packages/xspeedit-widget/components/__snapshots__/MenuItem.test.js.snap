// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem should render correctly when it's a link 1`] = `
<a
  className="jsx-3275355419 active"
  href="https://github.com/adbayb/xspeedit"
  target="_blank"
>
  <Icon
    height={24}
    name="github"
    width={24}
  />
  <JSXStyle
    css="a.jsx-3275355419{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex:1;-ms-flex:1;flex:1;cursor:pointer;-webkit-transition:background-color 500ms;transition:background-color 500ms;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.active.jsx-3275355419{background-color:#34b7b6;}"
    styleId="3275355419"
  />
</a>
`;

exports[`MenuItem should render correctly when it's active 1`] = `
<a
  className="jsx-3275355419 active"
  onClick={[Function]}
  role="button"
>
  <Icon
    height={24}
    name="github"
    width={24}
  />
  <JSXStyle
    css="a.jsx-3275355419{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex:1;-ms-flex:1;flex:1;cursor:pointer;-webkit-transition:background-color 500ms;transition:background-color 500ms;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.active.jsx-3275355419{background-color:#34b7b6;}"
    styleId="3275355419"
  />
</a>
`;

exports[`MenuItem should render correctly when it's not active 1`] = `
<a
  className="jsx-3275355419 "
  onClick={[Function]}
  role="button"
>
  <Icon
    height={24}
    name="github"
    width={24}
  />
  <JSXStyle
    css="a.jsx-3275355419{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex:1;-ms-flex:1;flex:1;cursor:pointer;-webkit-transition:background-color 500ms;transition:background-color 500ms;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.active.jsx-3275355419{background-color:#34b7b6;}"
    styleId="3275355419"
  />
</a>
`;
